library/Rules/CreditCard.php

```diff
@@ -11,18 +11,66 @@
 
 namespace Respect\Validation\Rules;
 
+use Respect\Validation\Exceptions\ComponentException;
+
 class CreditCard extends AbstractRule
 {
+    /**
+     * @var string
+     */
+    public $brand;
+
+    /**
+     * @var array
+     */
+    private $brands = [
+        'American Express' => '/^3[47]\d{13}$/',
+        'Diners Club' => '/^3(?:0[0-5]|[68]\d)\d{11}$/',
+        'Discover' => '/^6(?:011|5\d{2})\d{12}$/',
+        'JCB' => '/^(?:2131|1800|35\d{3})\d{11}$/',
+        'MasterCard' => '/^5[1-5]\d{14}$/',
+        'Visa' => '/^4\d{12}(?:\d{3})?$/',
+    ];
+
+    /**
+     * @param string $brand Optional credit card brand.
+     */
+    public function __construct($brand = null)
+    {
+        if (null !== $brand && !isset($this->brands[$brand])) {
+            $brands = array_keys($this->brands);
+            $message = sprintf('"%s" is not a valid credit card brand. Available: %s.', $brand, implode(', ', $brands));
```

Comment:
I liked what you did with the ComponentException message. For me it's fine now. :stuck_out_tongue: 

I thought about having constants for those brands, It would help just in case someone is using an IDE with autocomplete. But anyways I think it wouldn't add any value to the code.

Let's keep it as is.